### ISSUE TITLE
Feature: Atomic Server can store blobs on S3 compatible storage - like AWS S3, Cloudflare durable objects

### ISSUE THREAD

**Issue by AlexMikhalev** [atomicdata-dev/atomic-data-rust#543]
To maintain lightweight for Atomic Server give user the opportunity to add S3 credenitals and endpoint with option to synchronise binary data with external S3 compatible servers. 
I would check how  https://github.com/quickwit-oss/quickwit does it.

**Comment by joepio** [comment]
This would be pretty simple to implement, I think. The logic in `download.rs` and `upload.rs` could be altered depending on whether some `s3` opts (like keys and bucket ID) are being set.

**Comment by AlexMikhalev** [comment]
Yes, and QuickWit have example of downloading from S3 in multithreaded way.

**Comment by AlexMikhalev** [comment]
Prefered way to progress to use OpenDAL https://opendal.apache.org/docs/rust/opendal/services/struct.S3.html
